### DISCUSSION TITLE
Change splits `orientation`?

### DISCUSSION THREAD

**Discussion by OmiChrome** [mrjones2014/smart-splits.nvim#100] [Q&A] [answered]
change orientation like , split -> vsplit  , on key-strokes
vice-versa ,

any keybinds?! or any implementation in plugin

**Comment by mrjones2014** [comment]
It wouldn’t be too hard to do with a little bit of Lua. Just grab the buffer ID, close the vert split, open horizontal split, set original buffer ID in the new window.

**Comment by mrjones2014** [comment, marked as answer]
Something like this should work:

```lua
-- convert a horizontal split to a vertical split
local buf = vim.api.nvim_get_current_buf()
vim.cmd.q()
vim.cmd.vsp()
vim.api.nvim_win_set_buf(0, buf)
```

Though that wouldn't be "directional" like the plugin so maybe there is room for a plugin feature here.

**Reply by OmiChrome** [reply]
> there is room for a plugin feature here

im counting on that  , 
please , keep the issue open till the  feature is fully implemented  , Thanks!
